CMakeLists.txt

```diff
@@ -0,0 +1,21 @@
+cmake_minimum_required(VERSION 3.13.4)
+project(CMaize_Public_Depend2 VERSION 1.0.0 LANGUAGES CXX)
+
+include("${PROJECT_SOURCE_DIR}/cmake/get_cpp.cmake")
+
+cpp_find_or_build_dependency(
+    CMaizePublicDepend
+    #for testing & dev only - URL github.com/ally3999/CMaizePublicDepend
+    URL github.com/CMaizeExamples/CMaizePublicDepend2
+    BUILD_TARGET CMaizePublicDepend
+    FIND_TARGET CMaizePublicDepend::CMaizePublicDepend
+)
+
+cpp_add_library(
+    CMaizePublicDepend2
+    SOURCE_DIR "${CMAKE_CURRENT_LIST_DIR}"
+    INCLUDE_DIR "${CMAKE_CURRENT_LIST_DIR}"
+    DEPENDS CMakePublic CMaizePublicDepend
```

Comment:
The `CMakePublic` dependency is not a direct dependency of this project (i.e. we don't make calls to CMakePublic directly from CMaizePublicDepend2) so we shouldn't specify it here. It's the responsibility of CMaizePublicDepend's config file to propagate its public dependencies downstream. That said as we wrote it CMakePublic is a private dependency of CMaizePublicDepend (CMakePublic does not show up in CMaizePublicDepend's API). A quick and dirty way to make it a public dependency would be to add a templated function to CMaizePublicDepend along the lines of:

```.cpp
//cmake_public_depend.hpp
#pragma once
#include <cmake_public.hpp>

template<typename T>
int call_cmake_public_depend(T&& input) {
    return call_cmake_public() + std::forward<T>(input);
}

int call_cmake_public_depend();
```

From a C++ library design standpoint this is bad design, since we can still hide the call to CMakePublic simply by having the templated function convert the input to a known type, but let's ignore that fact :grin: